File: packages/gatsby-plugin-page-tree/src/menu-item.ts
export default class MenuItem {
    constructor() {
        this.path = "";
        this.title = "";
        this.selected = false;
        this.active = false;
        this.isEmptyParent = false;
        this.children = [];
    }
    path: string
    title: string
    selected: boolean
    active: boolean
    isEmptyParent: boolean
    children: Array<MenuItem>
}

File: src/DocGen.Web.Requirements/MenuItem.cs
using System.Collections.Generic;

namespace DocGen.Web.Requirements
{
    public class MenuItem
    {
        public MenuItem()
        {
            Children = new List<MenuItem>();
        }
        
        public string Title { get; set; }

        public string Path { get; set; }
        
        public bool Selected { get; set; }
        
        public bool Active { get; set; }
        
        public bool IsEmptyParent { get; set; }

        public List<MenuItem> Children { get; set; }
    }
}